### DISCUSSION TITLE
Add new entity category "maintenance"

### DISCUSSION THREAD

**Discussion by balloob** [home-assistant/architecture#1016] [Entity Models]
## Context

We have entity categories control, sensor, diagnostic and config. This allows us to group entities by purpose. There is a group of entities that are currently categorized as diagnostic that could benefit from a new category: "maintenance".

## Decision

Introduce a new category "maintenance". This category is for any entity that can indicate that maintenance is due for a device.

This category will be applied to battery sensors, but can also be used for things like "brush life left" entities from vacuum cleaners.

<img width="360" alt="Screenshot 2023-12-14 at 16 35 15" src="https://github.com/home-assistant/architecture/assets/1444314/e096846d-8da5-4759-a5fa-b2c90a2a41a5">

## Consequences

A bunch of diagnostic sensors and binary sensors will receive an updated category.

**Comment by allenporter** [comment]
Makes sense. Wondering if this make sense to be used by to-do entities... but that'd probably need to be user configurable via ui somehow, and my impression is that entity categories can't be configured that way today. Curious if you have thoughts about how to think about this?

**Reply by balloob** [reply]
To do is a special breed, as it's not really a device/service. Although a task could be for maintenance, it would feel weird to declare them as such, as we don't know that really in advance. I rather have a maintenance dashboard where one could pin a to do list.

**Reply by Madelena** [reply]
A to-do can be used as a chore for maintenance around the home, but it can also be used as something else. So not all to-dos are for maintenance, but some can be.

My recommendation would be to just let users pick what to-do lists they want to show on the Maintenance Dashboard.

**Comment by Madelena** [comment]
Having some sort of instructions attached to each entity that requires maintenance would be helpful, since almost everything that require maintenance has its own unique steps.

For example, for battery sensors, it would be helpful to include what type of battery needs to be replaced, so I can buy or charge the required batteries ahead of time.

For parts such as printer ink, vacuum brushes, and its sensors, it would be nice to have the links to the manufacturer FAQ for instructions, or even direct links to buy new parts.

**Reply by balloob** [reply]
I like this very much ❤️

**Reply by dxmnkd316** [reply]
I would strongly suggest Battery Notes in HACS.  It does much of this.  But I still like the idea of adding a maintenance category.  Would help with clearing up the sensors and diagnostic sections a bit.

**Comment by maffee** [comment]
I wasn't sure whether I should post this yet since I haven't had the time to check out the new to-do functionalities yet. But since I stumbled upon this thread today ... why not :) 

First of all, I very much like this idea of a maintenance category. I've had my own maintenance dashboard for ~6 years. It's always been somewhat difficult to build, though it got a bit easier when Helpers were introduced to HA. Maybe my experiences with the topic can provide some help to this.

I have a few main requirements & implementations for this. It's been pretty much the same for the whole 6 years (Janitor log was added ~2 years ago):

1. I need to be able to create "virtual" entities / maintenance targets for devices that have no sensors etc (oven or floor drain or terrace).
2. I need to be able to set a desired maintenance cycle ("wash / clean the floor drains every **90** days")
3. I need to be able to set the date the maintenance was last done
4. I need to be able to see how many days ago the last maintenance was last done
5. I need to be able to receive (repeating) notifications about a maintenace task about to be due (or is past due)
6. I need to be able to reset the maintenance counter (= set "maintenance was last done" to "today")
7. I need to be able to have a "janitor log" where I can see what was done and when, so that I can for example show it to a new owner of the house when I sell this (the reset button now writes a new line to a janitorlog.csv with a timestamp and text)


I have two UI's for this, one is more of a admin type that I use only through desktop and only sometimes. The more used view is a tablet view which is used from a wall mounted tablet. I have about ~10 different maintenance tasks at the moment (screenshot is only a partial view). The texts in the screenshots are in Finnish, sorry about that. Rough translations go like this:

"Suihkulasien pesu" = Shower closet's glass wall cleanup
1. input datetime "Suihkulasit pesty viimeksi" = Last cleanup/maintenance done
2. input number "Suihkulasien pesun kynnysarvo" = Maintenance cycle in days
3. template sensor "Suihkulasien pesusta aikaa" = Days since last maintenance was done
4. template sensor "Suihkulasien pesuun aikaa" = Days until next maintenance (I just noticed that this is quite funny when the due date is so much overdue...)
5. Gauge for days since last maintenance was done
6. Button "Nollaa" = Reset

Screenshot 1:
![screenshot1](https://github.com/home-assistant/architecture/assets/16758032/51cc161f-85b9-4053-9622-b4c4ac882d5f)

And the tablet view with less configs. Screenshot 2:
![screenshot2](https://github.com/home-assistant/architecture/assets/16758032/74f6043d-a73a-4f33-87ef-c6045b5f55b8)

**Comment by frenck** [comment]
We've discussed this proposal in today's architectural meeting and decided to close this one now.

While it might be viable, it might become less of a concern once we are done adding actual devices and device types to the Home Assistant concept. Therefore, we discussed closing it for now and re-evaluating in the future if needed.

../Frenck
